variable/type_conversion.cpp

```diff
@@ -20,8 +19,11 @@ struct MakeVariableWithType {
               [](const units::Unit &x) { return x; },
               [](const auto &x) {
                 if constexpr (is_ValueAndVariance_v<std::decay_t<decltype(x)>>)
-                  return ValueAndVariance<T>{static_cast<T>(x.value),
-                                             static_cast<T>(x.variance)};
+                  if constexpr (std::numeric_limits<T>::is_integer)
+                    return static_cast<T>(x.value);
```

Comment:
How can this be required? This looks like it is introducing a bug: Doesn't this now silently strip variances? Should maybe better throw an exception here, or check before calling transform.